src/main/java/com/gmail/yuramitryahin/dao/jdbc/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,101 @@
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String query = "SELECT * FROM manufacturer WHERE id=?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                return getManufacturer(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find manufacture by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String query = "SELECT * FROM manufacturer";
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                if (getManufacturer(resultSet).isPresent()) {
+                    manufacturers.add(getManufacturer(resultSet).get());
+                }
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find any manufacture", e);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturer SET name=?, country=? WHERE id=? AND deleted=false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.setLong(3, manufacturer.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacture " + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE manufacturer SET deleted=true WHERE id=?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacture by id " + id, e);
+        }
+    }
+
+    private Optional<Manufacturer> getManufacturer(ResultSet resultSet) throws SQLException {
```

Comment:
you need optional in one method, and the rest of the methods work directly with Manufactureres. I would suggest to return Manufacturers and not optional; but it's up to you

Reply:
I think this is necessary to avoid the NullPointerException

Reply:
because if there is no such id in the table we will receive null

Reply:
Я не наполягаю, тому й написав up to you

Reply:
Окей, просто пояснив свою точку зору)

Reply:
по суті цей метод в тебе приватний. Ти не дає доступ до нього іншим програмістам. Я погоджуюся, що тут краще повертати або об'єкт або null. Але в будь якому випадку в тебе тут null не буде при поточній реалізації. Відповідно Optional буде зайвим
```suggestion
    private Manufacturer getManufacturer(ResultSet resultSet) throws SQLException {
```

Reply:
fixed